server/src/main/java/org/elasticsearch/index/query/MatchPhraseQueryBuilder.java

```diff
@@ -173,9 +174,7 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         }
 
         MatchQuery matchQuery = new MatchQuery(context);
-        if (analyzer != null) {
-            matchQuery.setAnalyzer(analyzer);
-        }
+        matchQuery.setAnalyzer(analyzer);
```

Comment:
Why is this check not needed ? `setAnalyzer` will throw an exception if `analyzer` is null.